### ISSUE TITLE
Cannot successfully re-configure LDAP Auth. after upgrade to 2.1.1

### ISSUE THREAD

**Issue by Testicule** [ansible/awx#2884]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

 - API
 - UI

##### SUMMARY
Cannot configure LDAP connection to AD.

##### ENVIRONMENT
* AWX version: 2.1.1
* AWX install method: docker on linux
* Ansible version:  2.7.2
* Operating System: CentOS 7.5
* Web Browser: Chrome & Firefox & tower-cli

##### STEPS TO REPRODUCE
Configure LDAP connection
LDAP Bind DN: CN=adm_blah,OU=BLAH,DC=BLAH,DC=com
LDAP User DN Template: blank
LDAP Group Type: ActiveDirectoryGroupType
LDAP Require Group: blank
LDAP Deny Group: blank
LDAP Start TLS: off
LDAP User Search:
["OU=Service Management,DC=BLAH,DC=com","SCOPE_SUBTREE","(sAMAccountName=%(user)s)"]
LDAP Group Search:
["OU=Groups,DC=BLAH,DC=com","SCOPE_SUBTREE","(objectClass=group)"]
LDAP User Attribute Map:
{"first_name": "givenName","last_name": "displayName","email": "userPrincipalName"}
LDAP Group Type Parameters: blank
LDAP User Flags by Group:
{"is_superuser": ["CN=DGRP_BLAH,OU=Delegation,OU=Groups,DC=BLAH,DC=com"]}
LDAP Organisation Map:
{"AU": {"admins": false,"remove_admins": false,"remove_users": false,"users": true}}
LDAP Team Map:
{"Blah Team": {"organization": "AU","users": ["CN=DGRP_BLAH,OU=Delegation,OU=Groups,DC=BLAH,DC=com"],"remove": true}}

The LDAP Bind DN & password have been tested by doing an ldapsearch, which successfully returned results.

I've tried logging in as "blah\username", and "username @blah.com", and "username".
Nothing works.

This configuration was working in AWX 1.0.6.17.
No longer working in AWX 2.1.1

##### EXPECTED RESULTS
Successful login, or, some useful information in LOGS.

##### ACTUAL RESULTS
Login fails, and logs don't give anything more than this:
2018-12-07 07:01:15,932 DEBUG    django_auth_ldap search_s('OU=Service Management,DC=BLAH,DC=com', 2, '(sAMAccountName=%(user)s)') returned 0 objects: 
2018-12-07 07:01:15,932 DEBUG    django_auth_ldap Authentication failed for BLAH\adm_blah: failed to map the username to a DN.

**Comment by linuxphone** [comment]
I am having the same issue, can someone help on this regard.

**Comment by Testicule** [comment]
I was able to get it to work by deleting (from AWX) the AD users who had logged in in the past, prior to the upgrade. These users were obviously not exported and re-imported correctly by tower-cli. Although the import seemed to be successful, there was obviously something that got left behind. Perhaps AWX saves a kerberos ticket or something in the user profile, and this was not exported/reimported.

I found someone who had this exact same issue here:
  https://groups.google.com/forum/#!topic/awx-project/sEOqveIMNkQ

See comment by Eugene Archibald:
"Annoyingly in my case it had cached something in AWX from prior LDAP work that was causing logins to fail even after the LDAP was configured correctly. I removed the cached accounts and login worked properly."

Now I'm at AWX version 2, I will probably be using the awx-migrate tool for future updates, and as awx-migrate is able to migrate credentials etc, perhaps awx-migrate will successfully migrate AD users in future.

For now though, let this ticket be another heads-up to all grappling with these issues, that:
1) AWX needs better LDAP setup debugging. It needs a button to verify that the bind works. It needs a button to verify the other settings. And it needs some improved logging.
2) If you migrate your data with tower-cli, you'll need to delete AD users from AWX. They won't be able to login again until the user profile has been deleted.
